x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -518,6 +523,16 @@ private PreAnalysisResult receiveLookupIndexResolution(
             }
             return result.addLookupIndexResolution(index, lookupIndexResolution);
         }
+
+        if (lookupIndexResolution.get().indexNameWithModes().isEmpty() && lookupIndexResolution.resolvedIndices().isEmpty() == false) {
```

Comment:
In which scenario is this possible?

Reply:
> same scenario we're discussing

**Can you please, be specific?** (scenarios confusion led to long discussions and prolonging this PR merge)

In my mind there are [these three scenarios](https://github.com/elastic/elasticsearch/pull/132138#pullrequestreview-3075766660) plus the ones where the mapping of the lookup index is not empty but the query uses `keep v`. Plus those involving CCS.
Note: _Every scenario above should be tested with **local only indices** (no CCS) as well, btw (can be done in a follow up PR or opened an issue for it to be addressed later)._